pallets/order-book/src/benchmarking/mod.rs

```diff
@@ -0,0 +1,1942 @@
+            output_asset_id,
+            amount,
+            deduce_fee,
+        }
+    }
+}
+
+pub(crate) mod exchange_single_order_benchmark {
+    use super::*;
+    use crate::test_utils::create_and_fill_order_book;
+    use common::{balance, Balance, VAL, XOR};
+
+    pub struct Context<T: Config> {
+        pub caller: T::AccountId,
+        pub order_book_id: OrderBookId<AssetIdOf<T>, T::DEXId>,
+        pub expected_in: Balance,
+        pub expected_out: Balance,
+        pub caller_base_balance: Balance,
+        pub caller_quote_balance: Balance,
+    }
+
+    pub fn init<T: Config + trading_pair::Config>(_settings: FillSettings<T>) -> Context<T> {
+        let caller = alice::<T>();
+        frame_system::Pallet::<T>::set_block_number(1u32.into());
+
+        let order_book_id = OrderBookId::<AssetIdOf<T>, T::DEXId> {
+            dex_id: DEX.into(),
+            base: VAL.into(),
+            quote: XOR.into(),
+        };
+
+        create_and_fill_order_book::<T>(order_book_id);
+
+        assets::Pallet::<T>::update_balance(
+            RawOrigin::Root.into(),
+            caller.clone(),
+            order_book_id.base,
+            balance!(1000000).try_into().unwrap(),
+        )
+        .unwrap();
+
+        let caller_base_balance =
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.base, &caller).unwrap();
+        let caller_quote_balance =
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.quote, &caller).unwrap();
+        Context {
+            caller,
+            order_book_id,
+            expected_in: balance!(168.5),
+            expected_out: balance!(1685), // this amount executes only one limit order
+            caller_base_balance,
+            caller_quote_balance,
+        }
+    }
+
+    pub fn verify<T: Config + core::fmt::Debug>(_settings: FillSettings<T>, context: Context<T>) {
+        let Context {
+            caller,
+            order_book_id,
+            expected_in,
+            expected_out,
+            caller_base_balance,
+            caller_quote_balance,
+        } = context;
+
+        assert_last_event::<T>(
+            Event::<T>::MarketOrderExecuted {
+                order_book_id,
+                owner_id: caller.clone(),
+                direction: PriceVariant::Sell,
+                amount: OrderAmount::Base(expected_in.into()),
+                average_price: balance!(10).into(),
+                to: None,
+            }
+            .into(),
+        );
+        assert_eq!(
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.base, &caller).unwrap(),
+            caller_base_balance - expected_in
+        );
+        assert_eq!(
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.quote, &caller).unwrap(),
+            caller_quote_balance + expected_out
+        );
+    }
+}
+
+#[cfg(not(test))]
+pub use benchmarks_inner::*;
+
+#[cfg(not(test))]
+mod benchmarks_inner {
```

Comment:
This module is big enough, so I think you can move it into separate file. Also order book benchmarks now is big and complex, so I think we can also move it to separate crate as it's done for some another pallets

Reply:
I extracted the benchmarks into a separate crate. Whilst doing this the `lib.rs` file basically consists of `benchmarks_inner` module, so I left it there. Just to clarify, the module is needed to not compile the `benchmarks!` macro for tests; otherwise I'd put it without the module.

However, I couldn't manage to fix wasm compilation for `--tests` :( without tests it works, as well as with tests and not wasm. can't figure out the problem since I basically copied .toml from order-book :/
part of error message for context
```
error[E0152]: duplicate lang item in crate `std` (which `order_book_benchmarking` depends on): `panic_impl`
```

not sure how to fix though; maybe try to copy some benchmarking crate's toml?

Reply:
Fixed

Reply:
Also ran the benchmarks locally for the reference